gitlab/pkg/apis/sources/v1alpha1/gitlabsource_types.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2020 The TriggerMesh Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/apis/duck"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// Check that GitLabSource implements the Conditions duck type.
+var _ = duck.VerifyType(&GitLabSource{}, &duckv1.Conditions{})
+
+// GitLabSourceSpec defines the desired state of GitLabSource
+type GitLabSourceSpec struct {
+	// ServiceAccountName holds the name of the Kubernetes service account
+	// as which the underlying K8s resources should be run. If unspecified
+	// this will default to the "default" service account for the namespace
+	// in which the GitLabSource exists.
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// ProjectUrl is the url of the GitLab project for which we are interested
+	// to receive events from.
+	// Examples:
+	//   https://knative.dev/eventing-contrib/gitlab
+	// +kubebuilder:validation:MinLength=1
+	ProjectUrl string `json:"projectUrl"`
+
+	// EventType is the type of event to receive from Gitlab. These
+	// correspond to supported events to the add project hook
+	// https://docs.gitlab.com/ee/api/projects.html#add-project-hook
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:Enum=push_events,push_events_branch_filter,issues_events,confidential_issues_events,merge_requests_events,tag_push_events,note_events,job_events,pipeline_events,wiki_page_events
+	EventTypes []string `json:"eventTypes"`
+
+	// AccessToken is the Kubernetes secret containing the GitLab
+	// access token
+	AccessToken SecretValueFromSource `json:"accessToken"`
+
+	// SecretToken is the Kubernetes secret containing the GitLab
+	// secret token
+	SecretToken SecretValueFromSource `json:"secretToken"`
+
+	// SslVerify if true configure webhook so the ssl verification is done when triggering the hook
+	SslVerify bool `json:"sslverify,omitempty"`
+
+	// Sink is a reference to an object that will resolve to a domain
+	// name to use as the sink.
+	// +optional
+	Sink *corev1.ObjectReference `json:"sink,omitempty"`
```

Comment:
instead of ObjectReference we should use the Destination API, see:
https://github.com/knative/eventing-contrib/pull/684